models/optional_modules/core_web_vitals/scratch/bigquery/snowplow_web_vital_events_this_run.sql

```diff
@@ -0,0 +1,106 @@
+{{
+  config(
+    tags=["this_run"],
+    enabled=var("snowplow__enable_cwv", false) and target.type == 'bigquery' | as_bool(),
+    sql_header=snowplow_utils.set_query_tag(var('snowplow__query_tag', 'snowplow_dbt'))
+  )
+}}
+
+with prep as (
+
+  select
+    e.event_id,
+    e.event_name,
+    e.app_id,
+    e.platform,
+    e.domain_userid,
+    e.user_id,
+    e.page_view_id,
+    e.domain_sessionid,
+    e.collector_tstamp,
+    e.derived_tstamp,
+    e.dvce_created_tstamp,
+    e.load_tstamp,
+    e.geo_country,
+    e.page_url,
+    e.page_title,
+    e.useragent,
+
+    {{ snowplow_utils.get_optional_fields(
+        enabled=true,
+        fields=yauaa_fields(),
+        col_prefix='contexts_nl_basjes_yauaa_context_1_0_0',
+        relation=ref('snowplow_web_base_events_this_run'),
+        relation_alias='e') }},
+
+    {{ snowplow_utils.get_optional_fields(
+        enabled= true,
+        fields=[{'field': 'lcp', 'dtype': 'string'}, {'field': 'fid', 'dtype': 'string'}, {'field': 'cls', 'dtype': 'string'}, {'field': 'inp', 'dtype': 'string'}, {'field': 'ttfb', 'dtype': 'string'}, {'field': 'navigation_type', 'dtype': 'string'}],
+        col_prefix='unstruct_event_com_snowplowanalytics_snowplow_web_vitals_1_0_0',
+        relation=ref('snowplow_web_base_events_this_run'),
+        relation_alias='e') }}
+
+  from {{ ref("snowplow_web_base_events_this_run") }} as e
+
+  where {{ snowplow_utils.is_run_with_new_events('snowplow_web') }} --returns false if run doesn't contain new events.
+
+  and event_name = 'web_vitals'
+
+  and page_view_id is not null
+
+  -- exclude bot traffic
+  and not {{ snowplow_utils.get_field(column_name = 'contexts_com_iab_snowplow_spiders_and_robots_1_0_0',
+                            field_name = 'spider_or_robot',
+                            table_alias = 'e',
+                            type = 'boolean',
+                            array_index = 0)}} = True
+
+  {{ filter_bots() }}
+
```

Comment:
Wait does our filter bots not exclude bots flagged via that context?

Reply:
It does filter out specific keywords based on the useragent field. I think it should be more secure to do both?

Reply:
Yeah I agree!